### ISSUE TITLE
Changing one stream asks to "Reset all streams"" 

### ISSUE THREAD

**Issue by nataliekwong** [airbytehq/airbyte#32634]
If I'm only changing one stream's settings, I get asked to reset all my streams. I should only have to reset the streams I actually made changes to. We have the concept of "reset affected streams" when schema changes are detected. Loom:
https://www.loom.com/share/74ca9fb7f785410bb4a304d3b3a748f3

To reproduce:
- Go to a connection's replication settings
- Change the sync mode for a single stream and click "Save"
- See prompt to "reset all streams"

**Comment by pmossman** [comment]
Refining notes:
- Requires backend changes
- This is "by design" but now that we have capability for stream-level resets, we should rethink this
- Any dependency on Move enabling per-stream syncs?
  - Probably not, per-stream resets should be separate from per-stream syncs

**Comment by nataliekwong** [comment]
@flutra-osmani is this still in the backlog or something you are actively looking into?

This came up in the context of our breaking change migration guides, where we ask users to follow a set of steps (go to schema, reset all streams). Since we don't explicitly say that we'll only reset all the streams in the configuration, the user thinks we're asking them to reset all streams even if they aren't affected by the breaking change. This issue is preventing us from moving to more concise language in the migration guide that will help them understand the reduced radius of a breaking change.

**Comment by flutra-osmani** [comment]
@nataliekwong No, this is still in the backlog. We can re-focus on this issue and will also discuss this with Michael.

**Comment by nataliekwong** [comment]
Got it, in that case should we reassign you?

**Comment by flutra-osmani** [comment]
> Got it, in that case should we reassign you?

talked to Michael and he will take a look at this ticket.
